### PR TITLE
fix(core): Optimize `CleanupPendingOnDemandCachesAgent`

### PR DESCRIPTION
**Pull request by ajordens**
Switched to using `exists` vs `get` with smaller partitions.

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java [hunk 1]
```diff
@@ -88,33 +94,43 @@ void run(Collection<Provider> providers) {
...
        .collect(Collectors.toList());
 
-     List existingOnDemandKeys;
+     Map<String, Response<Boolean>> existingOnDemandKeys = new HashMap<>();
      if (redisClientDelegate.supportsMultiKeyPipelines()) {
-       existingOnDemandKeys = redisClientDelegate.withMultiKeyPipeline(pipeline -> {
-         onDemandKeys.forEach(k -> pipeline.get(provider.getProviderName() + ":onDemand:attributes:" + k));
-         return pipeline.syncAndReturnAll();
+       redisClientDelegate.withMultiKeyPipeline(pipeline -> {
+         for (List<String> partition : Iterables.partition(onDemandKeys, redisCacheOptions.getMaxDelSize())) {
```
Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: Could hardcode this to something small for the time being if we'd like -- it's not particularly critical

**Comment by ajordens** [comment]
Does this seem better @robzienert @asher ?

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java [hunk 2]
```diff
@@ -88,33 +94,43 @@ void run(Collection<Provider> providers) {
...
+           for (String id : partition) {
+             existingOnDemandKeys.put(id, pipeline.exists(provider.getProviderName() + ":onDemand:attributes:" + id));
```
Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: `exists` vs `get` must be better right?

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java (hunk 1 above)
Comment (asher) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: We currently have `maxDelSize` set to 2 which makes me think we can do away with pipelineing here entirely for simplicity's sake, given that this cleanup agent isn't time sensitive. Totally fine to leave as-is though, should be a nice improvement.

Comment (ajordens) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: I didn't look close enough to see what the maxes were set to in the various profiles ... it looked like `maxPipelineSize` was pretty high which is why I err'd towards `maxDelSize`.

I'll stick with it as-is and see how it goes.

**Pull request merged**